### ISSUE TITLE
Create and integrate light bulb (voltage) sounds

### ISSUE THREAD

**Issue by jbphet** [phetsims/faradays-law#163]
This is a child issue of #161.

We need a sound for the amount of light produced by the lightbulb, which corresponds to the amount of current being generated by moving the magnet.  A prototype that includes a sound for this can be seen at https://bayes.colorado.edu/dev/html/phet-io-wrapper-sonification/1.0.0-dev.126/build/phet-io-wrapper-sonification/faradays-law/html/faradays-law-sonification.html, but we should try not to be too influenced by it.

@Ashton-Morris will supply some ideas for timbre and chord layers and we'll review them at the next meeting as potential avenues to pursue before starting to actually integrate the sounds into the sim.

For reference, the current (and currently un-sonified) version of the sim can be found at https://phet.colorado.edu/sims/html/faradays-law/latest/faradays-law_en.html.

**Comment by Ashton-Morris** [comment]
These form a chord. Perhaps we can have them individually fade in. My thoughts are "chord-g3" 1% of meter to 100%, "chord-e4" 33% of meter to 100%, and "chord-c5" 66% of meter to 100%.  That way at 100% they all form a chord but at 25% you only hear one note.

[Here are all the files.](https://drive.google.com/drive/folders/1WHcnidTFlNuX8SmS6jtOl_8_2acf3yfR?usp=sharing)

- lightbulb-sound-chord-g3-loop.wav
- lightbulb-sound-chord-c5-loop.mp3
- lightbulb-sound-chord-c5-loop.wav
- lightbulb-sound-chord-e4-loop.mp3
- lightbulb-sound-chord-e4-loop.wav
- lightbulb-sound-chord-g3-loop.mp3


Here is a FM synthesis sounds to test if you may. It just gets triggered once when the meter starts to light up, its not a loop. One might be too low for computer speakers so I made a higher version as well. 

- lightbulb-sound-womp_high_version.mp3
- lightbulb-sound-womp_high_version.wav
- lightbulb-sound-womp_low_version.mp3
- lightbulb-sound-womp_low_version.wav

This is more literal but it was the first idea that came to my mind. Its the hum of a guitar cable. This should loop and I have [included a video showing how we could add a filter to follow the size of the light to make it more dynamic.](https://drive.google.com/file/d/13t0z5dKXZGw_VaDtFO0m9VyeY7h_1DhH/view?usp=sharing)

- lightbulb-sound-hum-loop.mp3
- lightbulb-sound-hum-loop.wav

**Comment by jbphet** [comment]
I integrated the `lightbulb-sound-chord` files listed above and we reviewed it in today's sound design meeting.  It's reasonably good, but we'd like to try for something a little less 'heavy'.  Words like 'brighter', 'shimmery', and 'sparkly' were used to describe what we think we're after.  For next week, we'd like to:

- [ ] Try some alternate sounds based on the words above.  @Ashton-Morris will create them, @jbphet will integrate.
- [ ] Try different notes for a little less of a major sound.  @Ashton-Morris will supply sounds that are based on fifths, @jbphet will integrate.
- [ ] @jbphet will try adding a filter that opens up as the voltage increases to try to match the 'brightness' of the bulb

**Comment by jbphet** [comment]
For reference, here is a dev version with the `lightbulb-sound-chord` sounds used for the light bulb:  https://phet-dev.colorado.edu/html/faradays-law/1.4.0-dev.57/phet/faradays-law_en_phet.html.

**Comment by Ashton-Morris** [comment]
Added some sounds to our folder. Still working on some other ideas.

- lightbulb-chord-idea-3-g5.mp3
- lightbulb-chord-idea-3-g5.wav
- lightbulb-chord-idea-3-c5.mp3
- lightbulb-chord-idea-3-c5.wav
- lightbulb-chord-idea-3-g4.mp3
- lightbulb-chord-idea-3-g4.wav
- lightbulb-chord-idea-3-c4.mp3
- lightbulb-chord-idea-3-c4.wav
- lightbulb-chord-idea-3-g3.mp3
- lightbulb-chord-idea-3-g3.wav
- lightbulb-chord-idea-2-g5.mp3
- lightbulb-chord-idea-2-g5.wav
- lightbulb-chord-idea-2-c5.mp3
- lightbulb-chord-idea-2-c5.wav
- lightbulb-chord-idea-2-g4.mp3
- lightbulb-chord-idea-2-g4.wav
- lightbulb-chord-idea-2-c4.mp3
- lightbulb-chord-idea-2-c4.wav
- lightbulb-chord-idea-2-g3.mp3
- lightbulb-chord-idea-2-g3.wav
- lightbulb-chord-idea-1-g5.mp3
- lightbulb-chord-idea-1-g5.wav
- lightbulb-chord-idea-1-c5.mp3
- lightbulb-chord-idea-1-c5.wav
- lightbulb-chord-idea-1-g4.mp3
- lightbulb-chord-idea-1-g4.wav
- lightbulb-chord-idea-1-c4.mp3
- lightbulb-chord-idea-1-c4.wav
- lightbulb-chord-idea-1-g3.mp3
- lightbulb-chord-idea-1-g3.wav

**Comment by Ashton-Morris** [comment]
Also I created a video of how it would sound with a filter on all of the tones. The first half of this video is without a filter and the second half is with a filter when the light gets brighter and dimmer.

[Video here.](https://drive.google.com/file/d/1Oyy70RZoNOQYfJJJ3pjKQfBSWrFJXyoa/view?usp=sharing)

**Comment by jbphet** [comment]
A duplicate issue was accidentally created in https://github.com/phetsims/faradays-law/issues/165.  I'm going to paste the comments from that issue below and then close it to avoid confusion.

@emily-phet said:
 >@Ashton-Morris @jbphet 
> 
> Please see the following link (to a section of the design doc) for my notes re designs for FL. The first section are my notes directly to you both re tomorrow's meeting (including some requests for ideas / sounds for @Ashton-Morris), and just below that is a summary of what I took from reviewing the demo videos and meeting notes from 2018.
https://docs.google.com/document/d/1_npYspPwkY1zrEOTUF_ITzTEZ_Qlra2i__4wap7pzP0/edit#heading=h.izqpmmc70elq

@Ashton-Morris said:

> I have created another example video showing a few ideas. Will cross post with our similar issue.
> 
> - I have a sine that fades in a saturated sine when the voltage meter is at 75% and above.
> - I have two chords that play for positive and negative that are more similar to what Mike had in his example.
> - I also show two different click sounds for when the voltage meter is at max.
> 
> [Video here.](https://drive.google.com/file/d/1VgmexOMw61LeE3eevu7xdrd7VpDXyzTU/view?usp=sharing)

@emily-phet said:

> @Ashton-Morris Thanks!
> 
> A few thoughts on variants:
> - I like the "smoothness" of the sine sound, and find this a relatively pleasant sine wave sound...but the second demo option is more "pleasant". Is there any way to find a sound that feels more like something between these two demos, where  the second demo sound sounds just a lit bit more synth-y? So it's loses a small amount of that recognizable-as-a-musical-instrument feel. If there's a way to try that out, I think it would be helpful. At the moment, I think the second demo sound is a good contender, and maybe this variant I'm proposing could be a little better. 
> - For the click sounds, I like the demo at 0:54 seconds. For that demo, is there a single click sound played twice, or is the click sound different for a positive max and negative max. If there the same, can you create a variant so we can try with one click sound for positive max, and a slightly different click sound for negative max?

**Comment by jbphet** [comment]
At today's sound design meeting we decided that the next steps are:

- [ ] @Ashton-Morris will create some sounds that we will try for the voltage that are based on the video he presented.  We need a root, major 3rd, perfect 5th, dominant 7th, major 7th, and octave, and plan to build a simple triad when the voltmeter is not showing and add in the octave or 7th based on whether the voltage is reading positive or negative when the meter *is* showing.
- [ ] @Ashton-Morris will supply a click sound for when the voltmeter hits a boundary.
- [ ] @jbphet will integrate these sounds once they are available.  The click sound will be played at a lower rate for one side versus the other, probably one step difference.

**Comment by Ashton-Morris** [comment]
[Here are the new files](https://drive.google.com/drive/folders/1WHcnidTFlNuX8SmS6jtOl_8_2acf3yfR?usp=sharing)
 
- voltage-max-click.wav
- lightbulb-voltage-chord-c5.mp3
- lightbulb-voltage-chord-c5.wav
- lightbulb-voltage-chord-b4.mp3
- lightbulb-voltage-chord-b4.wav
- lightbulb-voltage-chord-a-sharp-4.mp3
- lightbulb-voltage-chord-a-sharp-4.wav
- lightbulb-voltage-chord-g4.mp3
- lightbulb-voltage-chord-g4.wav
- lightbulb-voltage-chord-e4.mp3
- lightbulb-voltage-chord-e4.wav
- lightbulb-voltage-chord-c4.mp3
- lightbulb-voltage-chord-c4.wav
- voltage-max-click.mp3

**Comment by jbphet** [comment]
I integrated the latest sounds, and we reviewed them in today's sound design meeting.  Next steps:

- [ ] The sounds for the voltage reaching the max and min values was a bit more squishy and bassy than we were after, and @Ashton-Morris was a bit surprised by how they sounded in context.  Something may have happened during the export process.  @Ashton-Morris will create some new ones to try for next week.
- [ ] The voltage sound was reasonably we received, and we'd like to run it by the rest of the a11y team and ask them about the sound itself and how it works in relation to the voltmeter.  @jbphet will publish a dev version and ask for feedback.

**Comment by jbphet** [comment]
@jessegreenberg, @terracoda, @zepumph, and @Matthew-Moore240: The sound design team would like your input on an element of the sound design for Faraday's Law.  Can you please play around with the version linked below and answer these questions:

+ Do you think that the sound used to convey the voltage is reasonably effective and pleasing?  If you think something else might work better, please describe it.
+ The sounds used to indicate the voltage level change somewhat when the voltmeter is and isn't present.  Are the differences between the sounds played when the voltmeter present noticeable, and do you find the difference between positive and negative voltage values distinct?

Thanks.
https://phet-dev.colorado.edu/html/faradays-law/1.4.0-dev.58/phet/faradays-law_en_phet.html

**Comment by terracoda** [comment]
I think these sounds are really amazing!
- I think the voltage sound is pleasing, at least it is not at all annoying. I didn't have a feeling that I want to stop moving the magnet. 
- The volt meter sound is subtle, but I think it is discernible and distinguishable and the positive and negative tones are also distinguishable. I think that's saying a lot from me because I often don't hear the things that other people hear.

**Question:** Is any kind of stereo effect already implemented? If not, I bet some people would like, in addition to tonal or pitch differences, more negative tone in the left ear when the volt meter is negative and more positive sound in the right ear when the volt meter is in the positive range.

I like the magnet pick-up sound, though I think the magnet may need a visual change of some kind. I have the same issue with the Chemistry Book in Friction, and the balloons in BASE, and even the mass spheres in GFL and GFLB. There are many PhET objects that don't have any visual indication of being "hovered", "focused", or in some kind of "active" state with mouse and touch input. 

We have the solid pink line and the dashed pink line to indicate "focus" and "active" (or grabbed) states for keyboard/alternative input. It would be great to have a discussion on this issue.  

I really like the clanking against the coils. I noticed the smaller coil has a slightly different clank which is cool - higher tone/pitch - makes it seem smaller. 

I moved the magnet slowly and fast. I tried to carefully to examine what was happening with the volt meter when I flipped the polarity of the magnet, too. 

I think the sound design provides a lot of interesting focus and direction for the interactive description design.

I see there is no keyboard navigation, but that makes sense since the focus is currently on the sound design.

**Comment by terracoda** [comment]
Question 2: Is there any difference between the sound of flipping the magnet inside the small cool, and flipping the magnet inside the large coil?

There is a difference in the amount of light coming the light bulb, so I was wondering if there might be a small difference in duration or volume that I may not be hearing in these two constrasting cases?

**Comment by jessegreenberg** [comment]
I like the sound quite a bit, I think it is reasonably pleasing, I found the difference in sound with voltmeter visible to be distinguishable, as well as difference between positive/negative voltage.

Unrelated to above questions, I did notice just a little bit of distortion/crackling when the voltage is large. Perhaps it is just my crummy headphones, but thought I would mention.

**Comment by Ashton-Morris** [comment]
@jbphet I figured out the issue with the voltage max click file. I accidentally forgot to solo the track with the click on it so I exported any other sounds that were in other tracks along with the click. I have fixed it here. 

[voltage-max-click-fixed.mp3](https://drive.google.com/file/d/1Rw19DpK3L7aSLB-WL7FdMGA2vK-mcghd/view?usp=sharing)

**Comment by zepumph** [comment]
>Do you think that the sound used to convey the voltage is reasonably effective and pleasing? If you think something else might work better, please describe it.

Yes I like it a lot. I get Zelda ocarina of time water temple vibes. It feels really pleasant to me. I do think that it is possible to get tired of it. After a while it felt a bit like a child breathing through a harmonica was following me around. 
From a sound-only audit, I didn't like that the sound stopped so abruptly. I then began to realize that it was because it was following the model, and it shouldn't play when there is no model voltage. That said I wish it faded out a bit more.

>The sounds used to indicate the voltage level change somewhat when the voltmeter is and isn't present. Are the differences between the sounds played when the voltmeter present noticeable, and do you find the difference between positive and negative voltage values distinct?

I think that the sounds for positive and negative are both really nice. I think it makes sense to limit the sound when the voltmeter is hidden. Unfortunately I feel like this is an oversimplification. The sound here is really sonification, and the underlying model isn't changing, it is just that we aren't viewing a particular part of the model. The following use case seems really pedagogically foggy to me:
1. Load the sim, and it starts with a voltmeter. Use the sim a lot and learn/parse that there are two sounds, one for positive and one for negative charges.
2. Hide the voltmeter, now it sounds like the same actions yield different model results because you only get the sound for positive (or maybe a variation?). 

This is sound is sonification, and so the underlying model hasn't changed, so why should the sound? I could see a scaffolding-related argument potentially if the sim started with the voltmeter hidden, and you wanted the learner to build up their voltage-sound chops, but even then the argument feels a bit weak. 

On the other hand, sound is a view, just like the visuals, might it make sense to match those together?

-------

It wasn't asked for, but I thought I'd comment on the UI sounds.
1. I really like the sound when you run into the coils. I think you nailed the earcon (I think that's the one that tries to mimic a sound from the real world). It comes off a bit harsh though, perhaps sample banging on thicker bars, or maybe turn it down?
2. The pickup and drop sounds are really fun. I feel like I'm using the bar magnet to mine rocks in a quarry. I don't think this sound should stick around, but it very appealing!

------

I found a hard to reproduce bug where about 20% of the time when switching from this github issue tab back to the sim tab to play around again, the first click and drag on the bar magnet yielded no sound at first, and then sound kicked in after a couple of swipes through the coils. All subsequent interactions (after first drop) always have sound response for me.

Thanks for all the hard work.

**Comment by jbphet** [comment]
We listened to the integrated max voltage clicks in context during the 8.18.2020 sound design meeting and decided to try a version that is one octave lower for next week.  At that point, we think we can choose between them.  @Ashton-Morris will provide a revised sound and @jbphet will integrate.

**Comment by jbphet** [comment]
@jessegreenberg said:

> I did notice just a little bit of distortion/crackling when the voltage is large. Perhaps it is just my crummy headphones, but thought I would mention.

On what browser (and version of it) did you notice this?

**Comment by Matthew-Moore240** [comment]
_Do you think that the sound used to convey the voltage is reasonably effective and pleasing? If you think something else might work better, please describe it._

**I really like the voltage sound quite a bit. It for sure invokes the sound I would expect for it.** 

   

_The sounds used to indicate the voltage level change somewhat when the voltmeter is and isn't present. Are the differences between the sounds played when the voltmeter present noticeable, and do you find the difference between positive and negative voltage values distinct?_

**The differences in sound is noticeable but if I wasnt looking for it I think it would of taken me a little bit to notice but once its noticed its very easy to tell a difference. Same feeling with the positive and negative, once I knew there was a difference it was easy to notice it. also when the voltmeter is not present I cant hear any difference between the positive and negative if there is one at all.**

**Comment by jbphet** [comment]
@zepumph said:

> This is sound is sonification, and so the underlying model hasn't changed, so why should the sound?

This is a question that is very important to our thinking on this, and on sound design for PhET sims in general.  In this case, we plan to make a determination about whether the voltage sound should always be the same or vary based on whether the voltmeter is enabled will be decided based on feedback and interviews.

For next week, I'll add an option to the Sound Options Dialog that will make it possible to choose between the two modes easily.

@emily-phet - Assigning to you to elaborate on our thinking with regards to this question.

**Comment by emily-phet** [comment]
@zepumph 
Your question made me think about some of what we're learning we can do with sound and sonification. 

- We can add sound effects, so physical objects "sound physical"
- We can convey the underlying model through auditory means (straight up, traditional sonification)
- We can also provide different auditory "views" so-to-speak, in the same way we provide (in some sims) different visual views of the model. 

This last utility of auditory display is likely context dependent (same as visual) where the idea may make sense in all sims, but enacting it in a way that is perceptually understandable to many students may be too challenging given our timeframe and current knowledge of good design. We encounter this issue in visual design as well, when it comes to providing different visual "views" (or representations) of the model. You need a sound conceptual idea, and then to have that idea manifested in a way that is understandable to learners. In interviews, we'll learn if our current approach to Faraday's Law and providing different auditory views of voltage/current is interpretable by learners. What we learn will inform this sim, and future sims...but you're right, it's definitely an open question if our specific approach in this context will work.

**Comment by jessegreenberg** [comment]
> On what browser (and version of it) did you notice this?

I used Windows 10 Chrome Version 84.0.4147.125 (Official Build) (64-bit)

**Comment by jbphet** [comment]
@Ashton-Morris has created new max voltage sounds but noted the information about them in a different ticket, see https://github.com/phetsims/faradays-law/issues/166#issuecomment-679359200.

**Comment by jbphet** [comment]
The new max voltage click sounds are now integrated into the sim on master along with options for choosing between those and the previous version (or disabling them altogether).  I've also added options for selecting between some different behaviors for the highest note in the chord that is used to portray the voltage level.  We can review in today's sound design meeting (8/25/2020).

**Comment by jbphet** [comment]
We reviewed the sounds used for the max positive and max negative voltages (i.e. when the voltmeter "pegs") in the 8.25.2020 sound design meeting and decided to go with the lower click sound with two full musical steps between them (e.g. from Bb to D).  @jbphet will finalize this and remove the options associated with these sounds.

**Comment by jbphet** [comment]
We reviewed the behavior of the chord that is played to indicate voltage level (a.k.a. light bulb brightness) in the 8.25.2020 sound design meeting and decided to make it possible via a query parameter to set two of the possible modes so that we can interview on them and see how users respond.  The two modes are:
1. high note corresponds to voltmeter visibility
2. high note always on
The default should be the first one.  No need to support the "no high note" mode.

**Comment by jbphet** [comment]
In today's sound design meeting @Ashton-Morris, @emily-phet, and I decide to go with the top note playing all the time regardless of whether the voltmeter is visible.  This way it is consistent, and may create a question in the mind of the users (why is the sound different sometimes?) which they may then figure out once they make the voltmeter visible.

**Comment by jbphet** [comment]
The sounds are in place and have been internally reviewed quite a bit at this point.  I'm going to unassign the issue, but leave it open to make sure that we check these over at least one final time before publication.

**Comment by jbphet** [comment]
In today's sound design meeting we discussed whether we should consider panning the max voltage clicks left and right, and decided it's not worth the time and effort.

**Comment by emily-phet** [comment]
The sim is at the state for a final, final sound review. Any issues that come out of that, we'll create a new issue for.
